OpenRA.Mods.Common/Activities/LayMines.cs

```diff
@@ -60,6 +61,13 @@ public override bool Tick(Actor self)
 		{
 			returnToBase = false;
 
+			if (layingMine)
+			{
+				layingMine = false;
+				FinishLayingMine(self);
+				return false;
```

Comment:
this should not exit if theres no mine lay delay

Reply:
also, no mines should be layed if the activity was cancelled

Reply:
Really? It makes sense to me that at the start of the laying action, the mine is allocated ("unloaded") from the ammo pool and Minelayer starts laying it on the ground and arming it. Only after that is complete (with `PreLayDelay` > 0), the mine armed and ready for use. And since the mine has already been allocated from the ammo pool, it should not be possible to cancel the order.

In Earth 2140 (this PR was made with its minelayer's behavior in mind) the minelayer vehicle decreases ammo pool *after* the mine has been laid. However I believe better behavior is the one I suggested above.

Reply:
I suppose it depends on the feel of the game. Do you want your units to be responsive or clunky? For base OpenRA I think it makes more sense to have the minelaying action cancellable